invokeai/frontend/web/src/features/parameters/components/Parameters/Canvas/InfillAndScaling/ParamInfillMethod.tsx

```diff
@@ -36,6 +40,19 @@ const ParamInfillMethod = () => {
     [dispatch]
   );
 
+  useEffect(() => {
+    if (!appConfigData) return;
+    if (!appConfigData.patchmatch_enabled) {
+      const filteredMethods = infillMethods.filter(
+        (method) => method !== 'patchmatch'
+      );
+      dispatch(setAvailableInfillMethods(filteredMethods));
+      dispatch(setInfillMethod(filteredMethods[0]));
+    } else {
+      dispatch(setInfillMethod('patchmatch'));
+    }
+  }, [appConfigData, infillMethods, dispatch]);
+
```

Comment:
This is a good spot to leverage the listener middleware. Instead of "listening" for changes to the configuration data in the component with `useEffect()`, we can do tuck that logic away in the middleware.

I'll push a change to this PR implementing it like that.